Mathlib/Data/Matrix/Basic.lean

```diff
@@ -235,6 +235,20 @@ instance distribMulAction [Monoid R] [AddMonoid α] [DistribMulAction R α] :
 instance module [Semiring R] [AddCommMonoid α] [Module R α] : Module R (Matrix m n α) :=
   Pi.module _ _ _
 
+section ofLinearEquiv
+variable [Semiring R] [AddCommMonoid α] [Module R α]
+
+variable (R) in
+/-- `Matrix.of` as a `LinearEquiv`. -/
+def ofLinearEquiv : (m → n → α) ≃ₗ[R] Matrix m n α :=
+  LinearEquiv.refl _ _
```

Comment:
I would have gone for defining this in terms of `Matrix.of`, so we don't break through the `Matrix` definition barrier. (And we'd get the `apply` lemmas by `@[simps]`.) On the other hand I don't expect we'll be making `Matrix` an irreducible def / structure soon, so it might be worth having a simpler definition. I'll let you decide.

Reply:
I'll wait for https://github.com/leanprover-community/mathlib4/pull/17535 which defines more of these